File: day-6/part-1.py
with open('data.txt') as f:
  data = f.read().split('\n\n')

  sum = 0

  for group in data:
    answers = set()

    for passenger_answers in group.split():
      answers.update(passenger_answers)
    
    sum += len(answers)
  
  print(sum)

File: day-8/part-1.js
const { read } = require('../utils')

;(async () => {
  const data = await read('./data.txt')

  let accumulator = 0

  let i = 0

  const instructionMap = {}

  while (true) {
    if (instructionMap[i]) {
      console.log(accumulator)
      break
    }

    instructionMap[i] = true
    let [instruction, number] = data[i].split(' ')
    number = Number(number)

    if (instruction === 'nop') {
      i++
      continue
    }

    if (instruction === 'acc') {
      accumulator += number
      i++
      continue
    }

    if (instruction === 'jmp') {
      i += number
      continue
    }
  }
})()